<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>750</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Visionary</string>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="mouseTracking">
    <bool>true</bool>
   </property>
   <widget class="QLabel" name="whitebg">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>860</width>
      <height>680</height>
     </rect>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);
border-color: rgb(0, 0, 0);</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="bg_color">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>900</width>
      <height>731</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>711</width>
      <height>0</height>
     </size>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(183, 181, 179);</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="responseTime">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>630</y>
      <width>171</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255, 255, 255);</string>
    </property>
    <property name="text">
     <string>Response Time：0s</string>
    </property>
   </widget>
   <widget class="QLabel" name="rightPanel">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>0</y>
      <width>301</width>
      <height>731</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(64, 66, 68);
color: rgb(255, 255, 255);</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="mouseLocation">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>660</y>
      <width>251</width>
      <height>20</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255, 255, 255);</string>
    </property>
    <property name="text">
     <string>Mouse Location: [0, 0]</string>
    </property>
   </widget>
   <widget class="QLabel" name="colorBar">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>690</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="colorText">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>690</y>
      <width>151</width>
      <height>20</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255, 255, 255);</string>
    </property>
    <property name="text">
     <string>RGB:(255,255,255)</string>
    </property>
   </widget>
   <zorder>bg_color</zorder>
   <zorder>whitebg</zorder>
   <zorder>rightPanel</zorder>
   <zorder>mouseLocation</zorder>
   <zorder>colorBar</zorder>
   <zorder>responseTime</zorder>
   <zorder>colorText</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_as"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>编辑</string>
    </property>
    <addaction name="actionRecover"/>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
   </widget>
   <widget class="QMenu" name="menuFilter">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="title">
     <string>滤镜</string>
    </property>
    <widget class="QMenu" name="menuColorImageProcess">
     <property name="title">
      <string>彩色图像处理</string>
     </property>
     <addaction name="actionChannelSeperation"/>
     <addaction name="actionDecolor"/>
     <addaction name="actionadjustHSI"/>
     <addaction name="actionadjustGradation"/>
    </widget>
    <widget class="QMenu" name="menuAdjustContrast">
     <property name="title">
      <string>对比度调节</string>
     </property>
     <addaction name="actionEqualizeHistogram"/>
    </widget>
    <widget class="QMenu" name="menuSmoothFilter">
     <property name="title">
      <string>平滑滤波器</string>
     </property>
     <addaction name="actionBlur"/>
     <addaction name="actionSharpen"/>
    </widget>
    <widget class="QMenu" name="menuOperation">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>224</y>
       <width>182</width>
       <height>174</height>
      </rect>
     </property>
     <property name="title">
      <string>代数几何操作</string>
     </property>
     <addaction name="actionAdd"/>
     <addaction name="actionResize"/>
     <addaction name="actionResizeLinear"/>
     <addaction name="actionSpinNearest"/>
     <addaction name="actionSpinLinear"/>
    </widget>
    <widget class="QMenu" name="menuMorphological">
     <property name="title">
      <string>形态学操作</string>
     </property>
     <addaction name="actionErosion"/>
     <addaction name="actionDilation"/>
    </widget>
    <widget class="QMenu" name="menuDetectEdge">
     <property name="title">
      <string>边缘检测</string>
     </property>
     <addaction name="actionDetectEdgeSobel"/>
     <addaction name="actionDetectEdgeLaplacian"/>
     <addaction name="actionDetectEdgeCanny"/>
    </widget>
    <addaction name="actionBinarization"/>
    <addaction name="separator"/>
    <addaction name="menuColorImageProcess"/>
    <addaction name="menuAdjustContrast"/>
    <addaction name="menuSmoothFilter"/>
    <addaction name="menuDetectEdge"/>
    <addaction name="menuOperation"/>
    <addaction name="menuMorphological"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuFilter"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionNew">
   <property name="text">
    <string>新建(&amp;N)</string>
   </property>
   <property name="toolTip">
    <string>新建(N)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>打开(&amp;O)</string>
   </property>
   <property name="toolTip">
    <string>打开(O)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>保存为</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>退出</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>撤销</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>重做</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionDecolor">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>去色</string>
   </property>
   <property name="visible">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>关于</string>
   </property>
  </action>
  <action name="actionRecover">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>恢复</string>
   </property>
  </action>
  <action name="actionBinarization">
   <property name="text">
    <string>二值化</string>
   </property>
  </action>
  <action name="actionBlur">
   <property name="text">
    <string>高斯模糊</string>
   </property>
  </action>
  <action name="actionSharpen">
   <property name="text">
    <string>锐化</string>
   </property>
  </action>
  <action name="actionDilation">
   <property name="text">
    <string>膨胀</string>
   </property>
  </action>
  <action name="actionErosion">
   <property name="text">
    <string>腐蚀</string>
   </property>
  </action>
  <action name="actionChannelSeperation">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>通道分离</string>
   </property>
  </action>
  <action name="actionadjustHSI">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>HSI调节</string>
   </property>
  </action>
  <action name="actionadjustGradation">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>色阶调整</string>
   </property>
  </action>
  <action name="actionEqualizeHistogram">
   <property name="text">
    <string>直方图均衡化</string>
   </property>
  </action>
  <action name="actionAdd">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>加操作</string>
   </property>
  </action>
  <action name="actionResize">
   <property name="text">
    <string>缩放（最近邻）</string>
   </property>
  </action>
  <action name="actionResizeLinear">
   <property name="text">
    <string>缩放（线性插值）</string>
   </property>
  </action>
  <action name="actionSpinNearest">
   <property name="text">
    <string>旋转（最近邻）</string>
   </property>
  </action>
  <action name="actionSpinLinear">
   <property name="text">
    <string>旋转（线性插值）</string>
   </property>
  </action>
  <action name="actionDetectEdgeSobel">
   <property name="text">
    <string>Sobel边缘检测</string>
   </property>
  </action>
  <action name="actionDetectEdgeLaplacian">
   <property name="text">
    <string>拉普拉斯边缘检测</string>
   </property>
  </action>
  <action name="actionDetectEdgeCanny">
   <property name="text">
    <string>Canny边缘检测</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
